pkg/logging/factory.go

```diff
@@ -23,44 +22,29 @@ type ZapBuilder struct {
 
 //
 // Build new logger.
-func (b *ZapBuilder) New() (l logr.Logger) {
+func (b *ZapBuilder) New() (logger logr.Logger) {
 	var encoder zapcore.Encoder
-	var options []zap.Option
 	sinker := zapcore.AddSync(os.Stderr)
+	level := zap.NewAtomicLevelAt(zap.DebugLevel)
+	options := []zap.Option{
+		zap.AddStacktrace(zap.ErrorLevel),
+		zap.ErrorOutput(sinker),
+		zap.AddCallerSkip(1),
+	}
 	if Settings.Development {
 		cfg := zap.NewDevelopmentEncoderConfig()
 		encoder = zapcore.NewConsoleEncoder(cfg)
-		options = append(
-			options,
-			zap.Development(),
-			zap.AddStacktrace(zap.ErrorLevel))
+		options = append(options, zap.Development())
 	} else {
 		cfg := zap.NewProductionEncoderConfig()
 		encoder = zapcore.NewJSONEncoder(cfg)
-		options = append(
-			options,
-			zap.AddStacktrace(zap.WarnLevel),
-			zap.WrapCore(
-				func(core zapcore.Core) zapcore.Core {
-					return zapcore.NewSampler(
```

Comment:
Each `Sampler` creates a 2 dimensional array of int64 counters with a 5 MB footprint.  The sampler is designed to rate-limit writing the file.  MT creates a lot of _child_ loggers and I don't think the memory usage is worth the memory-bloat.